shotgun_api3/shotgun.py

```diff
@@ -675,6 +688,32 @@ def _construct_read_parameters(self,
         params["paging"] = { "entities_per_page": self.config.records_per_page,
                              "current_page": 1 }
 
+        if additional_server_presets is not None:
+            if not isinstance(additional_server_presets, list):
+                msg = "'additional_server_presets' parameter must be a list or None"
+                raise ValueError(msg)
+
+            preset_list = []
+            for preset in additional_server_presets:
+                if not isinstance(preset, list):
```

Comment:
Agree with having an example in the comments.  We'll obviously need to add this syntax into the docs on what the filters look like in the wiki.

As above, I'd just leave out the isinstance checking.  No reason to stop people from using their own classes to wrap this kind of syntax.